classes/local/store/object_file_system.php

```diff
@@ -298,11 +302,12 @@ public function delete_object_from_local_by_hash($contenthash, $objectsize = 0)
 
         if ($initiallocation === OBJECT_LOCATION_DUPLICATED) {
             $localpath = $this->get_local_path_from_hash($contenthash);
-
             if ($this->verify_external_object_from_hash($contenthash)) {
                 $success = unlink($localpath);
 
                 if ($success) {
+                    // Remove dir if empty.
+                    $this->delete_empty_folders(dirname($localpath));
```

Comment:
This only seems half right. Lets say we delete the last file eg

filedir/aa/bb/aabbccddeeff

now we are left with

fileedir/aa/bb/

then delete_empty_folders will delete the parent:

fileedir/aa/

but what about the grandparent? If it is empty we want to delete it too, but we should only attempt that if we deleted the first one, ie iterate up the ancestor tree. We should be careful to not iterate outside of the filedir though